calcom/cal.com
# fix: Gets Google Calendar working locally even when not using localhost:3000 domain to access the app

## What does this PR do?

The problem is as follows:
- Google Calendar(and most others integrations) don't allow adding a non `localhost` domain which is not secure and doesn't have a valid tLD. e.g app.cal.local has `.local` tLD which isn't a valid registered TLD. Also, even if we use a valid TLD, it would require you to use an HTTPS url which most people don't setup locally.
- But we can't use http://localhost:3000 for testing organizations.
- So, we use http://localhost:3000 only in Google Calendar and when setting up OAuth, we do that with the same domain but when adding credentials to DB, we redirect to http://app.cal.local so that the authentication is still possible.

Note that you an still work it around by running an ngrok tunnel but that changes it's URL on every restart and is not a good experience at all.


## Requirement/Documentation
We need to be able to test Google Calendar integration locally

## Type of change
  - [x] Bug fix (non-breaking change which fixes an issue) existing functionality to not work as expected)

## How should this be tested?
- Follow the steps "Obtaining the Google API Credentials" from README.md to be able to test Google Calendar locally.
	- 	Make sure to use http://localhost:3000 for redirect URLs whitelisting in Google Cloud.
- Access the app at app.cal.local:3000(which is a requirement when testing org stuff). Now try installing Google Calendar. You should be able to install it now. It isn't possible to do so in main,

## Mandatory Tasks


## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
